### DISCUSSION TITLE
Do you pay for egress transfer from Storage with Supabase Edge Functions?

### DISCUSSION THREAD

**Discussion by nthouliss** [supabase/supabase#7546] [Questions] [answered]
Hi, 

This question is somewhat related to #3403 but in relation to the Supabase Edge functions. 

I was wondering if the egress pricing for Supabase storage applies when you're using Supabase Edge Functions? For instance if I were to download a file to a Supabase edge funciton, would I have to pay for the egress transfer?

Any insight would be appreciated :)

**Comment by burggraf** [comment, marked as answer]
Yes, I believe that would count towards your egress transfer, as it doesn't matter where the transfer is initiated from.

**Reply by nthouliss** [reply]
Darn, was hoping there might be some networking magic for decreased or free transfer within the network. Thanks!

**Reply by siobh9** [reply]
Dang, this is a bummer and very well might force me to move storage to AWS where data egress is free between S3 and all other AWS services within a region - would otherwise much much rather use Supabase Storage and Edge Functions. Is there any chance this has changed @burggraf?

**Reply by inian** [reply]
This is the still the same. We are currently working on supporting R2 for our storage and reducing our egress through that - regardless of if the object is being downloaded from edge functions or not.

**Reply by siobh9** [reply]
Oh awesome!! Look forward to following along!
